packages/webdriver/src/until-url-available.ts

```diff
@@ -0,0 +1,23 @@
+import { Operation, spawn, timeout } from 'effection';
+import { fetch, Response } from './fetch';
+
+/**
+ * An operation that completes when the server at `url` begins
+ * returning successful responses to an HTTP GET request.
+ */
+export function* untilURLAvailable(url: string, maxWait: number): Operation<void> {
+  yield spawn(function* () {
+    yield timeout(maxWait);
+    throw new Error(`timed out waiting ${maxWait}ms for ${url} to become available`)
+  });
+  while (true) {
+    try {
+      let response: Response = yield fetch(url);
+      if (response.ok) {
+        return;
+      }
+    } catch (error) {
+      /* This is ok. Error is raised whenever a network connect fails */
```

Comment:
Can we be more specific about this? We probably don't want to fetch *any* error, just a network error. Would be better to rethrow if we get a different error.

Reply:
Now we check to see if it is a `FetchError` which is raised whenever the network conditions prevent a fetch from being performed.